### DISCUSSION TITLE
Cache issue

### DISCUSSION THREAD

**Discussion by KalaiJay** [mockoon/mockoon#953] [Help] [answered]
Hi ,

I am testing multiple request and response using Mockoon. I have set rules in Mockoon for each request and response. I am sending a request from Postman. When I send request, I notice the rules whatever I set of the requests are getting cached and returning the same value in the next subsequent responses. For example below are the requests I am sending one after other.
//Request 1

{
	"fruits": [{
		"name": "apple",
		"quantity": 30.000,
		"stores": [{
			"storeId": "01",
			"storename": "walmart"
		},
{			"storeId": "02",
			"storename": "kmart"
		}]
	}]
}

//Request2

{
	"fruits": [{
		"name": "grapes",
		"quantity": 30.000,
		"stores": [{
			"storeId": "01",
			"storename": "walmart"
		}
	}]
}

The rule I set in Mockoon is to look for a fruit name and return a response based on that fruit. If I run first request , it sends the below response

//Response

{
 
    "fruits": [
        {
            "name": "apple",
            "quantity": 30,
            "stores": [
                {
                    "storeId": "01",
                    "storename": "walmart",
                    "fruitprice": 120
                },
                {
                    "storeId": "02",
                    "storename": "kmart",
                    "fruitprice": 140
                }
            ]
        }
    ]
}

Whereas, if I send a second response with fruit as grape, I still get the fruit name as apple in the response , and the store information's are populating a blank value.  Looks like some cache is happening.  Can someone help me on this.

Note:

I tried to clear cache in postman and in Mockoon as well by passing the Cache-Control in the header as no-store. But still see the rule field/property is getting cached in the subsequent responses

**Comment by 255kb** [comment, marked as answer]
Could you please share your setup? (https://mockoon.com/docs/latest/mockoon-data-files/environment-clipboard-copy/)
Without seeing the rules you applied it's a bit hard to help.

**Reply by KalaiJay** [reply]
This how i am setting my rules. 
![Mockoon](https://user-images.githubusercontent.com/108901937/220676460-b42053b3-8c58-44c2-b9e4-c0e99bd0c13d.PNG)

**Reply by KalaiJay** [reply]
Also, I tried without the double quotes in the rules (apple). still getting the same response for apple instead of grapes. Looks like response is getting cached. If stop and start the Mockoon for every request and test it, it works fine.

**Reply by 255kb** [reply]
yes, you need to remove the quotes in the value. 

Normally, there is no caching. So, two more questions: 
- Is the request sent with an application/json Content-type? It's required to be able to look into the body with a path like `fruits.0.name`.
- Are you using a [Data bucket](https://mockoon.com/docs/latest/data-buckets/overview/) as the response's body? Data buckets are persisting and generated at server start or during the first call if they access request's values (like with the `body` helper). Which means their content will always stay the same. If that's the issue, I suggest that you use an inline body instead.

**Reply by KalaiJay** [reply]
Thank you for your information, After changing the response body to have as an line instead of Data bucket  cache issue got resolved.

**Reply by 255kb** [reply]
Great! (it's not a cache issue it's a feature 🙂)
